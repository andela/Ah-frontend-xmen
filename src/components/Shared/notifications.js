import React from 'react';


export const UserDropdownMenu = props => (
  <ul className={`dropdown ${props.active}`}>
    <li><a href="/editor">Create an Article</a></li>
    <li><a href="/profile">My Profile</a></li>
    <li>
      <a
        id="logout"
        onClick={props.logOut}
        role="button"
        tabIndex="0"
      >
        Log out
      </a>
    </li>
  </ul>
);


const NotificationsDropdownMenu = (props) => {
  const notifications = props.notifications.slice();
  if (notifications.length > 4) {
    notifications.splice(4);
    notifications.push({
      message: 'see more',
      actionLink: '/notifications',
    });
  }
  return (
    <ul className={`note-dropdown ${props.active}`}>
      {notifications.map((notification, index) => (
        <li
          key={`note_${index}`}
          className={
          !notification.isUnread ? 'notification-item notification-unread'
            : 'notification-item'
        }
        >
          <a
            href="#"
            onClick={() => (props.markAsRead(notification.actionLink))}
          >
            {notification.message}
          </a>
        </li>
      ))
      }

    </ul>
  );
};

export default NotificationsDropdownMenu;
