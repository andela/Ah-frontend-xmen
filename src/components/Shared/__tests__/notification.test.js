import React from 'react';
import { shallow } from 'enzyme';
import NotificationsDropDownMenu from '../notifications';
import { UserDropdownMenu } from '../notifications';


describe('NotificationsDropdownMenu', () => {
  it('should render many notifications without crashing', () => {
    const notifications = [];
    for (let i = 0; i < 9; i += 1) {
      notifications.push({
        message: 'notifcation message',
      });
    }
    const props = {
      notifications,
    };
    const wrapper = shallow(<NotificationsDropDownMenu {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render many user dropdown without crashing', () => {
    const props = {};
    const wrapper = shallow(<UserDropdownMenu {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
