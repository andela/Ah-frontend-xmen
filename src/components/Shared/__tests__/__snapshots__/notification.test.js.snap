// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationsDropdownMenu should render many notifications without crashing 1`] = `ShallowWrapper {}`;

exports[`NotificationsDropdownMenu should render many user dropdown without crashing 1`] = `ShallowWrapper {}`;
